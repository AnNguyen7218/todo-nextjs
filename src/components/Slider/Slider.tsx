import React from 'react';
import Swiper, {
  ReactIdSwiperProps,
  ReactIdSwiperChildren,
  GetSwiper,
} from 'react-id-swiper';
import 'swiper/css/swiper.css';

const swiperDefaultParams: ReactIdSwiperProps = {
  loop: true,
  slidesPerView: 'auto',
  grabCursor: false,
  centeredSlides: false,
};

type Props = {
  swiperParams?: ReactIdSwiperProps;
  children: ReactIdSwiperChildren;
  getSwiper?: GetSwiper;
};

const Slider: React.FC<Props> = ({
  swiperParams = {},
  children,
  getSwiper,
}) => {
  const mergedSwiperParams = { ...swiperDefaultParams, ...swiperParams };
  return (
    <Swiper {...mergedSwiperParams} getSwiper={getSwiper}>
      {children}
    </Swiper>
  );
};

export default Slider;
