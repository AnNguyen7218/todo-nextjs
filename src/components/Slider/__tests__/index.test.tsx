import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Slider from '../Slider';

afterEach(cleanup);

describe('Slider Component', () => {
  test('Slider should match snapshot', () => {
    const { container } = render(
      <Slider>
        <div>Slide 1</div>
        <div>Slide 2</div>
        <div>Slide 3</div>
        <div>Slide 4</div>
      </Slider>
    );
    expect(container).toMatchSnapshot();
  });
});
