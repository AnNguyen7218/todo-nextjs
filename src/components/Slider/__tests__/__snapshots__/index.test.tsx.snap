// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component Slider should match snapshot 1`] = `
<div>
  <div
    class="swiper-container swiper-container-initialized swiper-container-horizontal"
  >
    <div
      class="swiper-wrapper"
      style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        Slide 1
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="1"
      >
        Slide 2
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="2"
      >
        Slide 3
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="3"
      >
        Slide 4
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        Slide 1
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        Slide 2
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="2"
      >
        Slide 3
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="3"
      >
        Slide 4
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        Slide 1
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="1"
      >
        Slide 2
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="2"
      >
        Slide 3
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="3"
      >
        Slide 4
      </div>
    </div>
    <span
      aria-atomic="true"
      aria-live="assertive"
      class="swiper-notification"
    />
  </div>
</div>
`;
